src/osmpbf.rs

```diff
@@ -81,104 +78,79 @@ pub struct BlockIndex {
     pub blob_len: usize,
 }
 
-struct BlockIndexIterator {
-    reader: BufReader<File>,
+struct BlockIndexIterator<'a> {
+    data: &'a [u8],
     cursor: usize,
-    file_buf: Vec<u8>,
-    is_open: bool,
 }
 
 enum BlobInfo {
     Header(BlockIndex),
     Unknown(usize, usize, Vec<u8>),
 }
 
-impl BlockIndexIterator {
-    fn new<P: AsRef<Path>>(path: P) -> io::Result<Self> {
-        let file = File::open(path)?;
-        Ok(Self {
-            reader: BufReader::new(file),
-            cursor: 0,
-            file_buf: Vec::new(),
-            is_open: true,
-        })
+impl<'a> BlockIndexIterator<'a> {
+    fn new(data: &'a [u8]) -> Self {
+        Self { data, cursor: 0 }
+    }
+
+    fn read(&mut self, len: usize) -> &[u8] {
```

Comment:
Result here and a length check?

Reply:
Or you could just use a Read (implemented for &u8) as a member?

Reply:
Is this really necessary? It will panic in any case. Read is not enough since I am using the cursor state somewhere outside.

Reply:
I was just thinking about what a user sees on the command line: error message or random panic. I think previously we were throwing errors all the way up?

Reply:
Actually this is a private function in an iterator, so we kind of have a guarantee here that it never panicks since it is only used in the iterator impl.